data-processing/common/commands/fetch_s2_data.py

```diff
@@ -44,41 +85,49 @@ def load(self) -> Iterator[ArxivId]:
             yield arxiv_id
 
     def process(self, item: ArxivId) -> Iterator[S2Metadata]:
-        # XXX(andrewhead): S2 API does not have versions of arXiv papers. I don't think this
-        # will be an issue, but it's something to pay attention to.
-        versionless_id = self._strip_arxiv_version(item)
-        resp = requests.get(
-            f"https://api.semanticscholar.org/v1/paper/arXiv:{versionless_id}"
-        )
 
-        if resp.ok:
-            data = resp.json()
-            references = []
-            for reference_data in data["references"]:
-                authors = []
-                for author_data in reference_data["authors"]:
-                    authors.append(Author(author_data["authorId"], author_data["name"]))
-                reference = Reference(
-                    s2_id=reference_data["paperId"],
-                    arxivId=reference_data["arxivId"],
-                    doi=reference_data["doi"],
-                    title=reference_data["title"],
-                    authors=authors,
-                    venue=reference_data["venue"],
-                    year=reference_data["year"],
-                )
-                references.append(reference)
-            
-            if not references:
-                # References are required to process citations, mark job as failed
-                raise S2ReferencesNotFoundException()
-
-            s2_metadata = S2Metadata(s2_id=data["paperId"], references=references)
-            logging.debug("Fetched S2 metadata for arXiv paper %s", item)
-            yield s2_metadata
+        try:
+            resp = self._mk_api_request(item)
+
+        except Exception:
+            # Exceptions here could reflect temporary service availability
+            # or slowdown issues with S2's API.
+            raise S2ApiException()
```

Comment:
do we maybe want to capture what the exception was and log it? Right now it's swallowed?

Reply:
Yeah sure, seems reasonable.